deployment/clouddeploy/deployer.go

```diff
@@ -357,18 +468,161 @@ func (p *Deployer) buildModifySpecs(
 	return specs, nil
 }
 
-func (p *Deployer) NewCluster(ctx context.Context, def *clusterdef.Cluster) (deployment.ClusterInfo, error) {
-	var clusterVersion string
-	for _, nodeGroup := range def.NodeGroups {
-		if clusterVersion == "" {
-			clusterVersion = nodeGroup.Version
+func (p *Deployer) DeployNewCluster(ctx context.Context, def *clusterdef.Cluster, clusterVersion string, serverImage string) (deployment.ClusterInfo, error) {
```

Comment:
Why do you pass the clusterVersion/serverImage separately here? They should be part of the def I think?